src/main/diagnostics/abstractCodeActionProvider.ts

```diff
@@ -3,13 +3,13 @@ import { ExtensionComponent } from '../extensionComponent';
 import { DependenciesTreeNode } from '../treeDataProviders/dependenciesTree/dependenciesTreeNode';
 import { TreesManager } from '../treeDataProviders/treesManager';
 import { Severity, SeverityUtils } from '../types/severity';
-import { DiagnosticsUtils } from './diagnosticsUtils';
 
 /**
  * @see DiagnosticsManager
  */
 export abstract class AbstractCodeActionProvider implements vscode.CodeActionProvider, ExtensionComponent {
     static readonly XRAY_DIAGNOSTIC_SOURCE: string = 'JFrog Xray';
+    private _gutterDecorations: vscode.TextEditorDecorationType[] = [];
```

Comment:
Looks like this variable is not in use, let's push the gutter decorations to this array and make sure to clear this array on text document close.
On Maven multi-project, it doesn't show decorations on submodules. This can be the root cause of the issue.